internal/rule/rule.go

```diff
@@ -0,0 +1,28 @@
+package rule
+
+type Rule struct {
```

Comment:
yeaaah thought about it. I wanted to make it so that other things can be checked too :(

Reply:
I'm okay with keeping it. It just seems very specific to permissions, just by the input being a string. Not sure what this can be extended to, or if it's worth doing so

Reply:
changed a bit